src/StoreBase.ts

```diff
@@ -60,178 +56,211 @@ export abstract class StoreBase {
+                        newMeta.throttledUntil = Math.min(throttledUntil, existingMeta.throttledUntil);
+                    }
+                    if (existingMeta && existingMeta.bypassBlock) {
+                        newMeta.bypassBlock = true;
+                    }
+                    StoreBase._pendingCallbacks.set(sub.callback, newMeta);
                 });
         } else {
+            const keys = _.map(_.isArray(keyOrKeys) ? keyOrKeys : [keyOrKeys], key => _.isNumber(key) ? key.toString() : key);
             // Key list, so go through each key and queue up the callback
             _.forEach(keys, key => {
                 _.forEach(this._subscriptions[key], callback => {
-                    const existingKeys = this._gatheredCallbacks.get(callback);
-                    if (existingKeys === undefined) {
-                        this._gatheredCallbacks.set(callback, [key]);
-                    } else if (existingKeys === null) {
+                    const existingMeta = StoreBase._pendingCallbacks.get(callback);
+                    StoreBase._updateExistingMeta(existingMeta, throttledUntil, bypassBlock);
+                    if (existingMeta === undefined) {
+                        StoreBase._pendingCallbacks.set(callback, { keys: [key], throttledUntil, bypassBlock });
+                    } else if (existingMeta.keys === null) {
                         // Do nothing since it's already an all-key-trigger
                     } else {
                         // Add it to the end of the list
-                        existingKeys.push(key);
+                        existingMeta.keys.push(key);
                     }
                 });
 
                 _.forEach(this._autoSubscriptions[key], sub => {
-                    const existingKeys = this._gatheredCallbacks.get(sub.callback);
-                    if (existingKeys === undefined) {
-                        this._gatheredCallbacks.set(sub.callback, [key]);
-                    } else if (existingKeys === null) {
+                    const existingMeta = StoreBase._pendingCallbacks.get(sub.callback);
+                    StoreBase._updateExistingMeta(existingMeta, throttledUntil, bypassBlock);
+                    if (existingMeta === undefined) {
+                        StoreBase._pendingCallbacks.set(sub.callback, { keys: [key], throttledUntil, bypassBlock });
+                    } else if (existingMeta.keys === null) {
                         // Do nothing since it's already an all-key-trigger
                     } else {
                         // Add it to the end of the list
-                        existingKeys.push(key);
+                        existingMeta.keys.push(key);
                     }
                 });
             });
 
             // Go through each of the all-key subscriptions and add the full key list to their gathered list
             _.forEach(this._subscriptions[StoreBase.Key_All], callback => {
```

Comment:
[Note: This is not exactly copy/paste from above, but it is close. Perhaps it is worth considering a helper function for most of the common logic?]